src/Analysis/Engine/Impl/OverloadResult.cs

```diff
@@ -117,44 +112,54 @@ public class OverloadResult : IOverloadResult {
 
                 return res;
             });
-            var returnType = overloads.SelectMany(o => o.ReturnType).Distinct();
 
+            var returnType = overloads.SelectMany(o => o.ReturnType).Distinct();
             return new OverloadResult(parameters, name, doc, returnType);
         }
 
         public override string ToString() {
-            return "{0}({1})->[{2}]{3}".FormatInvariant(
-                Name,
-                string.Join(",", Parameters.Select(p => "{0}{1}:{2}={3}".FormatInvariant(p.Name, p.IsOptional ? "?" : "", p.Type ?? "", p.DefaultValue ?? ""))),
-                string.Join(",", ReturnType.OrderBy(k => k)),
-                string.IsNullOrEmpty(Documentation) ? "" : ("'''{0}'''".FormatInvariant(Documentation))
+            var parameters = string.Join(", ",
```

Comment:
Where do we use this `ToString` implementation? In serialization?

Reply:
It actually generates the signature